<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CannySettings</class>
 <widget class="QWidget" name="CannySettings">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>829</width>
    <height>398</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CannySettings</string>
  </property>
  <widget class="QGroupBox" name="groupBox">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>571</width>
     <height>181</height>
    </rect>
   </property>
   <property name="title">
    <string>VOI преобразование</string>
   </property>
   <property name="flat">
    <bool>false</bool>
   </property>
   <property name="checkable">
    <bool>false</bool>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_7">
    <property name="leftMargin">
     <number>6</number>
    </property>
    <property name="topMargin">
     <number>6</number>
    </property>
    <property name="rightMargin">
     <number>6</number>
    </property>
    <property name="bottomMargin">
     <number>6</number>
    </property>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_7">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Window Center</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="centerVal">
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="singleStep">
           <number>5</number>
          </property>
          <property name="stepType">
           <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="spacing">
         <number>6</number>
        </property>
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <property name="sizeConstraint">
           <enum>QLayout::SetFixedSize</enum>
          </property>
          <item>
           <widget class="QSpinBox" name="centerMin">
            <property name="buttonSymbols">
             <enum>QAbstractSpinBox::NoButtons</enum>
            </property>
            <property name="minimum">
             <number>-65535</number>
            </property>
            <property name="maximum">
             <number>65536</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_7">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Минимум</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QSlider" name="centerSlider">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <property name="sizeConstraint">
           <enum>QLayout::SetMinimumSize</enum>
          </property>
          <item>
           <widget class="QSpinBox" name="centerMax">
            <property name="buttonSymbols">
             <enum>QAbstractSpinBox::NoButtons</enum>
            </property>
            <property name="minimum">
             <number>-65535</number>
            </property>
            <property name="maximum">
             <number>65536</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_8">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Максимум</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Window Width</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="widthVal"/>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <property name="spacing">
         <number>6</number>
        </property>
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <property name="sizeConstraint">
           <enum>QLayout::SetFixedSize</enum>
          </property>
          <item>
           <widget class="QSpinBox" name="widthMin">
            <property name="buttonSymbols">
             <enum>QAbstractSpinBox::NoButtons</enum>
            </property>
            <property name="minimum">
             <number>-65535</number>
            </property>
            <property name="maximum">
             <number>65536</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_9">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Минимум</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QSlider" name="widthSlider">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <property name="sizeConstraint">
           <enum>QLayout::SetMinimumSize</enum>
          </property>
          <item>
           <widget class="QSpinBox" name="widthMax">
            <property name="buttonSymbols">
             <enum>QAbstractSpinBox::NoButtons</enum>
            </property>
            <property name="minimum">
             <number>-65535</number>
            </property>
            <property name="maximum">
             <number>65536</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_10">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Максимум</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QGroupBox" name="groupBox_2">
   <property name="geometry">
    <rect>
     <x>600</x>
     <y>130</y>
     <width>131</width>
     <height>119</height>
    </rect>
   </property>
   <property name="title">
    <string>Размытие Гаусса</string>
   </property>
   <property name="flat">
    <bool>false</bool>
   </property>
   <property name="checkable">
    <bool>false</bool>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Размер ядра свёртки</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spinBox"/>
      </item>
      <item>
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Сигма</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDoubleSpinBox" name="doubleSpinBox">
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.050000000000000</double>
        </property>
        <property name="stepType">
         <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
        </property>
        <property name="value">
         <double>1.000000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QGroupBox" name="groupBox_3">
   <property name="geometry">
    <rect>
     <x>590</x>
     <y>10</y>
     <width>131</width>
     <height>119</height>
    </rect>
   </property>
   <property name="title">
    <string>Порог Canny</string>
   </property>
   <property name="flat">
    <bool>false</bool>
   </property>
   <property name="checkable">
    <bool>false</bool>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Window Center</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEdit_3">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>20</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Window Width</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEdit_4">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>20</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>centerMin</sender>
   <signal>valueChanged(int)</signal>
   <receiver>CannySettings</receiver>
   <slot>setVOICenterMinimum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>44</x>
     <y>76</y>
    </hint>
    <hint type="destinationlabel">
     <x>77</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>centerMax</sender>
   <signal>valueChanged(int)</signal>
   <receiver>CannySettings</receiver>
   <slot>setVOICenterMaximum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>557</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>546</x>
     <y>205</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widthMin</sender>
   <signal>valueChanged(int)</signal>
   <receiver>CannySettings</receiver>
   <slot>setVOIWidthMinimum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>57</x>
     <y>155</y>
    </hint>
    <hint type="destinationlabel">
     <x>83</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widthMax</sender>
   <signal>valueChanged(int)</signal>
   <receiver>CannySettings</receiver>
   <slot>setVOIWidthMaximum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>538</x>
     <y>156</y>
    </hint>
    <hint type="destinationlabel">
     <x>537</x>
     <y>234</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>centerSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>centerVal</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>115</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>118</x>
     <y>46</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>centerVal</sender>
   <signal>valueChanged(int)</signal>
   <receiver>centerSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>106</x>
     <y>39</y>
    </hint>
    <hint type="destinationlabel">
     <x>174</x>
     <y>79</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widthVal</sender>
   <signal>valueChanged(int)</signal>
   <receiver>widthSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>108</x>
     <y>124</y>
    </hint>
    <hint type="destinationlabel">
     <x>178</x>
     <y>164</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widthSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>widthVal</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>157</x>
     <y>158</y>
    </hint>
    <hint type="destinationlabel">
     <x>124</x>
     <y>126</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setVOICenterMinimum(int)</slot>
  <slot>setVOICenterMaximum(int)</slot>
  <slot>setVOIWidthMinimum(int)</slot>
  <slot>setVOIWidthMaximum(int)</slot>
 </slots>
</ui>
